pkg/executor/builder.go

```diff
@@ -3997,7 +3999,7 @@ func (b *executorBuilder) buildIndexMergeReader(v *plannercore.PhysicalIndexMerg
 // 1. dataReaderBuilder calculate data range from argument, rather than plan.
 // 2. the result executor is already opened.
 type dataReaderBuilder struct {
-	plannercore.Plan
+	plan plannercore.Plan
```

Comment:
As `IndexLookUpExecutor` embeded the `dataReaderBuilder` directly, it shouldn't embed `Plan` to avoid calling the `ID` of the `Plan` unexpectedly.